src/client-agent/start_agent.c

```diff
@@ -12,234 +12,150 @@
 #include "agentd.h"
 #include "os_net/os_net.h"
 
+#ifdef UNIT_TESTING
+    #define static
+    #ifdef WIN32 
+            #include "unit_tests/wrappers/client-agent/start_agent.h"
+            #undef CloseSocket
+            #define CloseSocket wrap_closesocket
+            #define recv wrap_recv
+    #endif
+#endif
+
 #define ENROLLMENT_RETRY_TIME_MAX   60
 #define ENROLLMENT_RETRY_TIME_DELTA 5
 
 int timeout;    //timeout in seconds waiting for a server reply
 
-static ssize_t receive_message_udp(const char *msg, char *buffer, unsigned int max_lenght);
-static ssize_t receive_message_tcp(const char *msg, char *buffer, unsigned int max_lenght);
+static ssize_t receive_message(char *buffer, unsigned int max_lenght);
 static void w_agentd_keys_init (void);
+static bool agent_handshake_to_server(int server_id);
+static void send_msg_on_startup(void);
 
-/* Attempt to connect to all configured servers */
-int connect_server(int initial_id)
+/**
+ * @brief Connects to a specified server
+ * @param server_id index of the specified server from agt servers list
+ * @retval true on success
+ * @retval false when failed
+ * */
```

Comment:
We should add a post-condition like:
```c
* @post The remote IP id (rip_id) is set to server_id if and only if this function succeeds.
```

Reply:
Changed in last commit but for some reason it is still not showing up in here